internal/handler/msg_createuser.go

```diff
@@ -108,65 +103,39 @@ func (h *Handler) MsgCreateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 
 	common.Ignored(document, h.L, "writeConcern", "authenticationRestrictions", "comment")
 
-	defMechanisms := must.NotFail(types.NewArray("SCRAM-SHA-1", "SCRAM-SHA-256"))
+	allMechanisms := must.NotFail(types.NewArray("SCRAM-SHA-1", "SCRAM-SHA-256"))
 
-	mechanisms, err := common.GetOptionalParam(document, "mechanisms", defMechanisms)
+	mechanisms, err := common.GetOptionalParam(document, "mechanisms", allMechanisms)
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
-	var credentials *types.Document
+	if mechanisms.Len() == 0 {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"mechanisms field cannot be empty",
+		)
+	}
 
+	var password string
 	if document.Has("pwd") {
-		pwdi := must.NotFail(document.Get("pwd"))
-		pwd, ok := pwdi.(string)
+		pwd := must.NotFail(document.Get("pwd"))
+		_, ok := pwd.(string)
 
 		if !ok {
 			return nil, handlererrors.NewCommandErrorMsg(
 				handlererrors.ErrTypeMismatch,
 				fmt.Sprintf("BSON field 'createUser.pwd' is the wrong type '%s', expected type 'string'",
-					handlerparams.AliasFromType(pwdi),
+					handlerparams.AliasFromType(pwd),
 				),
 			)
 		}
 
-		credentials, err = makeCredentials(mechanisms, username, pwd)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	id := uuid.New()
-	saved := must.NotFail(types.NewDocument(
-		"_id", dbName+"."+username,
-		"credentials", credentials,
-		"user", username,
-		"db", dbName,
-		"roles", types.MakeArray(0),
-		"userId", types.Binary{Subtype: types.BinaryUUID, B: must.NotFail(id.MarshalBinary())},
-	))
-
-	adminDB, err := h.b.Database("admin")
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
-
-	users, err := adminDB.Collection("system.users")
-	if err != nil {
-		return nil, lazyerrors.Error(err)
+		password = pwd.(string)
```

Comment:
If you call this something other than `password` you won't have import issue, `userPassword`?